### ISSUE TITLE
DVR/NEWS support

### ISSUE THREAD

**Issue by wahlmanj** [iBaa/PlexConnect#530]
Is DVR, lite tv, and news working in the newest release? If not, is it planned? Glad to see you joined the team moody :)

**Comment by moodyblue** [comment]
Hi Jim, glad to see that you are still around. The fast answer is no, these are not in plan.

I'm not a developer myself, but as you are, you're welcome to develop new functions. I can interface with Plex Inc to gather the how/to's.

**Comment by david-nedved** [comment]
+1

**Comment by david-nedved** [comment]
@wahlmanj - Just an update, I've got my Live TV/DVR working through the [IPTV](https://github.com/Cigaras/IPTV.bundle) plugin in an m3u playlist - Basically my TV Tuner (a HDHomerun), is feeding into [tvheadend](https://hub.docker.com/r/linuxserver/tvheadend/) which is then feeding into Plex via [tvhproxy](https://hub.docker.com/r/chvb/docker-tvhproxy/).

The plus side of doing it this way, is I've actually added in various additional IPTV Feeds (BBC One, NBC, CBS etc) as well as my local Australian Free to air channels (via my HDHomerun) and I can get accurate EPG data across all of the channels by manually specifying the EPG file in Plex (By default in Plex, you're restricted to a single geographic region which sucks).

It would be awesome getting native DVR added into the PlexConnect plugin, but for now the IPTV Channel plugin does the trick - There's also a better plugin called (LiveTVH)[https://github.com/taligentx/LiveTVH.bundle] which should in theory work better than the IPTV plugin, but it's currently got some issues with PlexConnect I'm trying to work through.

I know it's not a brilliant fix, but it works - hope this helps you on your endeavours.

**Comment by moodyblue** [comment]
DVR support has been implemented, credits to @SheaSmith. Can you please test ?

**Comment by sreknob** [comment]
Great news @moodyblue - going to give it a spin.
I think this calls for a new release!

**Comment by wahlmanj** [comment]
Ill test as soon as I get home, awesome work!

**Comment by sreknob** [comment]
Just tried here. Not having any love. The interface shows (and it’s nice!) but attempting to play anything results in “Trailers not available”. Will get logs and report back.

**Comment by moodyblue** [comment]
I'm having the same problem. I've already reported to @SheaSmith, the developer

**Comment by SheaSmith** [comment]
Should be resolved in #556

**Comment by sreknob** [comment]
Getting closer! Now able to get to start the stream but doesn’t get past showing a still image of the current frame. If I wait, then exit and restart the image updates to a new one and the blue bar shows between the two time points but no playable video, even when trying to scrub. Nothing in the info logs, will turn on debug and report back. FWIW, I’m using an HDHR network tuner.

**Comment by moodyblue** [comment]
Probably a codec issue. Are you able to see the stream details with plex
web ?

A sábado, 23 de nov de 2019, 14:49, sreknob <notifications@github.com>
escreveu:

**Comment by SheaSmith** [comment]
Yes, that's definitely a codec issue. If you could just send a screenshot of what appears in the dashboard when playing from Plex Web. e.g.
![image](https://user-images.githubusercontent.com/51303984/69484734-860dea00-0e9b-11ea-8860-8d6b84c51b1d.png)

Because this is a bunch of new codecs as far as PlexConnect is concerned I might have to add it to the list of those that need transcoded.

On a side note, I'm tempted to rewrite the transcode engine to be more of a whitelist, rather than a blacklist, but it's quite old code, and I'm very cautious about making sure nothing breaks.

**Comment by moodyblue** [comment]
@SheaSmith I can handle the transcoding engine

@sreknob Can you pls try with Transcoder Action = Transcode in both ATV's ?

**Comment by moodyblue** [comment]
@SheaSmith I have several `films` and `tv shows` libraries, but PlexConnect does not allows me to pick the right one when I want to record.

**Comment by SheaSmith** [comment]
Yeah, there are currently no options with regards to recording - it just does it based on the defaults. Doing the options is a bit tricky since there's a lot of pieces that have to be put together (e.g. pulling all film/TV libraries), plus the limitations of the ATV (e.g. no dropdown dialogs).

**Comment by wahlmanj** [comment]
So far it works great. Fantastic work thus far! is there a way to get a list of channels similar to the atv4 plex app that would have a similar channel guide like you would see on a normal tv provider?

**Comment by SheaSmith** [comment]
No, that's not possible. We're very limited with regards to UI components, unfortunately.

**Comment by wahlmanj** [comment]
![image](https://user-images.githubusercontent.com/5344121/69490453-e37b4a00-0e4d-11ea-8743-2c8b3244a277.jpg)

**Comment by wahlmanj** [comment]
Not possible? Interesting...

**Comment by SheaSmith** [comment]
Oh, you meant something like that? I thought you were referring to the more traditional grid guide:
![image](https://user-images.githubusercontent.com/51303984/69490483-93ec5b00-0eed-11ea-95d0-c8dca30555fa.png)
The 'On Now' page could be themed like the image you have attached, but the problem is whether it replaces the On Now page (which would make it inconstitant with Plex web) or whether its just an option in settings.

**Comment by wahlmanj** [comment]
Well that would be preferable and fantastic but something like the image above will suffice as well.

**Comment by moodyblue** [comment]
I would prefer to continue this conversation in https://forums.plex.tv/t/live-tv-dvr/490437/3

This can facilitate the contact eith Plex team, if needed, and also contribute to my scorecard.

**Comment by ENjOY88** [comment]
HDHR.Viewer2 would be a great solution to this if it can somehow be built into PlexConnect Live TV.

**Comment by SheaSmith** [comment]
Live TV support has already been added, just not news yet, so this issue hasn't been closed

**Comment by ghokun** [comment]
Not directly related to this topic, but people in this topic (maybe the only ones in the world) would be interested so I decided to comment here:

I wrote an application just like PlexConnect for IPTV playlists.

It can be found here: https://github.com/ghokun/appletv3-iptv

I can remove this comment if it is inappropriate.

**Comment by moodyblue** [comment]
> 
> 
> Not directly related to this topic, but people in this topic (maybe the only ones in the world) would be interested so I decided to comment here:
> 
> I wrote an application just like PlexConnect for IPTV playlists.
> 
> It can be found here: https://github.com/ghokun/appletv3-iptv
> 
> I can remove this comment if it is inappropriate.
Great. Good luck for your app.
